src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2821,245 +2821,176 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
-                {
-                    REQUIRE(fabs(ltx2.getPrefetchHitRate() - 1.0f) < 0.0001f);
-                }
-                REQUIRE(numLoadedHot == 0);
-            }
-            else
-            {
-                REQUIRE(fabs(ltx2.getPrefetchHitRate() - 1.0f) > 0.0001f);
-                if (expectedSuccessKeys.size() > 0)
-                {
-                    REQUIRE(numLoadedHot >=
-                            expectedSuccessKeys.size() - cacheSize);
-                }
-            }
+        auto preLoadPrefetchHitRate = root.getPrefetchHitRate();
+        REQUIRE(preLoadPrefetchHitRate == 0);
 
-            // Ensure that the meter is exhausted after loading.
-            // i.e. not all expectedSuccessKeys have remaining capacity.
-            // canLoad should return false for at least some keys in
-            // expectedSuccessKeys now.
-            ensureLedgerKeyMeterCanLoad(lkMeterHot, allEntries,
-                                        /*success keys*/ {},
-                                        /* fail keys */ expectedSuccessKeys);
-        };
-
-        std::vector<LedgerEntry> allEntries;
-        // Overlapping, all keys succeed
-        SECTION("all keys have quota")
-        {
-            // Make sure entries is mix of classic soroban
-            auto tx1Entries =
-                std::vector<LedgerEntry>{entries.at(0), entries.at(1)};
-            auto tx2Entries =
-                std::vector<LedgerEntry>{entries.at(0), entries.at(1)};
-            addTxn(0, tx1Entries);
-            addTxn(-1, tx2Entries);
-            allEntries.insert(allEntries.end(), tx1Entries.begin(),
-                              tx1Entries.end());
-            allEntries.insert(allEntries.end(), tx2Entries.begin(),
-                              tx2Entries.end());
-            checkPrefetch(allEntries);
-        }
-
-        SECTION("dead keys don't affect quota")
-        {
-            auto deadKeys =
-                std::vector<LedgerKey>{deadKeySet.begin(), deadKeySet.end()};
-            auto tx1Entries =
-                std::vector<LedgerEntry>{entries.at(0), entries.at(1)};
-            auto tx2Entries =
-                std::vector<LedgerEntry>{entries.at(0), entries.at(1)};
-            addTxn(0, tx1Entries);
-            addTxn(-1, tx2Entries, deadKeys);
-            allEntries.insert(allEntries.end(), tx1Entries.begin(),
-                              tx1Entries.end());
-            allEntries.insert(allEntries.end(), tx2Entries.begin(),
-                              tx2Entries.end());
-            checkPrefetch(allEntries);
-        }
-
-        SECTION("don't load entries without quota")
-        {
-            auto tx1Entries =
-                std::vector<LedgerEntry>{entries.at(0), entries.at(1)};
-            addTxn(-1, tx1Entries);
-            checkPrefetch(tx1Entries);
+        // Ensure that the correct entries were loaded.
+        for (auto const& k : expectedSuccessKeys)
```

Comment:
Sorry I missed this last iteration, but we don't actually need this loop at all. The hotCache assert with 100% hit rate is sufficient to show that the first prefetch loaded all entries correctly.

Reply:
If I don't call `ltx2.load(k)`, then the hit rate will be zero as nothing has been loaded. I guess I don't need the assertion tho.

Reply:
Ah, I thought the prefetch itself contributed to the hit rate, but I guess it's just loads after the prefetch, thanks!